drivers/serial/uart_mcux.c

```diff
@@ -449,3 +477,4 @@ static const struct uart_mcux_config uart_mcux_##n##_config = {		\
 	UART_MCUX_INIT_CFG(n);
 
 DT_INST_FOREACH_STATUS_OKAY(UART_MCUX_INIT)
+PM_DEVICE_DT_INST_DEFINE(0, uart_mcux_pm_action);
```

Comment:
you are defining it but not binding to the driver instance. Define it inside the `UART_MCUX_INIT` loop and bind it to the instance. Basically, on `DEVICE_DT_INST_DEFINE` instead of `NULL`  you pass `PM_DEVICE_DT_INST_GET(n)`